src/NuGet.Core/NuGet.Commands/MSBuildProjectFactory.cs

```diff
@@ -149,11 +152,23 @@ private void AddOutputLibFiles(IEnumerable<OutputLibFile> libFiles, HashSet<stri
             }
         }
 
-        private void AddFileToBuilder(ManifestFile packageFile)
+        private bool AddFileToBuilder(ManifestFile packageFile)
         {
             if (!Files.Any(p => packageFile.Target.Equals(p.Target, StringComparison.CurrentCultureIgnoreCase)))
             {
-                Files.Add(packageFile);
+                var fileExtension = Path.GetExtension(packageFile.Source);
+
+                if(IncludeSymbols &&
+                    PackArgs.SymbolPackageFormat == SymbolPackageFormat.Snupkg &&
+                    !string.Equals(fileExtension, ".pdb", PathUtility.GetStringComparisonBasedOnOS()))
+                {
```

Comment:
It may be worth adding a comment as to why the `StringComparison` matters here depending on the OS.

Reply:
it doesn't actually. i'll change to ordinalignorecase.